src/util.c

```diff
@@ -1375,12 +1452,20 @@ common_dir_prefix_length(const char *s1, const char *s2)
 {
 	const char *p1 = s1;
 	const char *p2 = s2;
+    char separator = '\0';
```

Comment:
Why bother init to 0? When you omit the default value you will get compiler warnings when the variable is being used without proper initialization. Also it can be made const via
`const char separator = strchr(p1, '/') ? '/' : '\\';`